src/live-programming.rst

```diff
@@ -20,9 +20,331 @@ Data Analysis Environments
 Debugging
 =========
 
+  "The EDSAC was on the top floor of the building and the tape-punching
+  and editing equipment one floor below... It was on one of my journeys
+  between the EDSAC room and the punching equipment that...the realization
+  came over me with full force that a good part of the remainder of my
+  life was going to be spent in finding errors in my own programs."
+
+  -- *Memoirs of a Computer Pioneer*, Maurice Wilkes
+
+Much of the activity of programming involves **debugging**, i.e.,
+diagnosing and preventing undesired program behavior.
+Wilkes's early realization of this fact was echoed and amplified in a 2002
+study of the U.S. software industry, which found that software engineers
+spent an average of 70-80% of their time testing and debugging, with
+the average bug taking 17.4 hours to fix.
```

Comment:
this is an unexpectedly large number, might need to dig into this / provide some context

Reply:
Not a surprising number imo. I definitely spent 2-3 days on several bugs as an engineer. Makes more sense when you consider the time it takes to reproduce the bug, which may be super dependent on a user's data, which you then might have to check out an encrypted copy of on a remote server, debug remotely (which is slow), repeat long repro steps each time you wanna investigate something new, etc.

I took out the mysterious 70-80% figure.